### DISCUSSION TITLE
Weird `undefined` message with 0.60.3

### DISCUSSION THREAD

**Discussion by ilg-ul** [mesonbuild/meson#10188] [Q&A]
I upgraded my build environment from 0.59.x to 0.60.3 and the builds pass, but I get a weird message:

```console
...
Program qemu-system-gnuarmeclipse found: YES (/Users/ilg/My Files/WKS Projects/micro-os-plus.github/xPacks/micro-test-plus-xpack.git/build/stm32f4discovery-meson-release/xpacks/.bin/qemu-system-gnuarmeclipse)
Build targets in project: 8

micro-os-plus-micro-test-plus undefined. <--- ???

  User defined options
    Cross files  : /Users/ilg/My Files/WKS Projects/micro-os-plus.github/xPacks/micro-test-plus-xpack.git/xpacks/micro-os-plus-build-helper/meson/toolchains/arm-none-eabi-gcc.ini
                   /Users/ilg/My Files/WKS Projects/micro-os-plus.github/xPacks/micro-test-plus-xpack.git/tests/platform-stm32f4discovery/meson/cross.ini
    backend      : ninja
    buildtype    : minsize
    platform-name: stm32f4discovery

Found ninja-1.10.2 at '/Users/ilg/My Files/WKS Projects/micro-os-plus.github/xPacks/micro-test-plus-xpack.git/xpacks/.bin/ninja'
...
```

`micro-os-plus-micro-test-plus` is the project name:

```meson
project('micro-os-plus-micro-test-plus',
  ['c', 'cpp'],
  default_options: [
    'c_std=c11',
    'cpp_std=c++20',
    'b_staticpic=false', # Mandatory, startup fails otherwise.
  ],
  meson_version: '>= 0.59'
)
```

There are no other references to this string in the project.

Any idea what the message is trying to tell?

**Comment by xclaesse** [comment]
Undefined is the version of your projects.

**Reply by ilg-ul** [reply]
The version of my project is clearly defined in `package.json`. 

For CMake I know that it is possible to parse JSON files, but for meson how can I read it from there and pass it to the `project()` command?

**Reply by xclaesse** [reply]
I don't think there is a built-in way of doing that, but you can do with an external python script for example `version : run_command('get-version.py', check : true).stdout().strip()`. You would have to write `get-version.py` that parse the json and print version to stdout.

**Reply by ilg-ul** [reply]
Thank you for the suggestion. Way more complicated than for CMake, so I'll probably have to get used with the undefined... :-(
